shopfloor/services/checkout.py

```diff
@@ -823,17 +887,20 @@ def list_dest_package(self, picking_id, selected_line_ids):
         lines = self.env["stock.move.line"].browse(selected_line_ids).exists()
         return self._response_for_select_dest_package(picking, lines)
 
-    def _set_dest_package_from_selection(self, picking, selected_lines, package):
+    def _set_dest_package_from_selection(
+        self, picking, selected_lines, package, barcode=None
+    ):
         if not package:
-            return self._response_for_select_dest_package(picking, selected_lines)
+            return self._response_for_select_dest_package(
+                picking,
+                selected_lines,
+                message=self.msg_store.package_not_found_for_barcode(barcode),
```

Comment:
what if barcode is none?

Reply:
probably you should not call `_set_dest_package_from_selection` if you don't have a package

Reply:
I'll check this point.

Reply:
Yes it's a bit hackish, but this happens only when `_set_dest_package_from_selection` is called by `set_dest_package` (which has a `package_id` as parameter), so the only to have `barcode=None` here is we send a wrong `package_id` to `set_dest_package`, and then yes the error message is not really appropriate (I talked about this in the related test).
Maybe the best way here is to return the generic "record not found" error message before calling `_set_dest_package_from_selection` yes.

Reply:
Fixed, we now test that the package exists in `set_dest_package` and `scan_dest_package` before calling `_set_dest_package_from_selection`. The error message differs between the two methods if the package can't be found.